### ISSUE TITLE
Tracking issue for nested issues and tasks list from the oscal-cli nodejs repo

### ISSUE THREAD

**Issue by avdouglas-nist** [usnistgov/OSCAL#1113]
This is a tracking issue for the remaining tasks for issues [#7](https://github.com/usnistgov/oscal-cli-nodejs/issues/7): Rewrite CI/CD pipeline content converter and schema generation and [#5 ](https://github.com/usnistgov/oscal-cli-nodejs/issues/5): Rewrite CI/CD pipeline copy and convert content in the oscal-cli-nodejs repository:  https://github.com/avdouglas-nist/oscal-cli-nodejs/tree/feature-build-setup.

- [x] Address accumulator issue (usnistgov/metaschema#180)
- [ ] Address entity parsing issue (usnistgov/oscal-cli-nodejs#9) @aj-stein-nist
- [x] Revise the fromFormat and toFormat options
- [ ] Generate-Schema [#7](https://github.com/usnistgov/oscal-cli-nodejs#7)
  - [ ] Update parameters in the Saxon JS API to perform the schema generation transformation
  - [x] Make module consistent with other modules
  - [x] Allow user to input (XML), and a specified output to produce the JSON (.json) or XML (.xsd) schema
  - [x] Save output to user specified location
  - [x] Add `--validate` option to validate the produced schema
  - [x] Add fromFormat and toFormat options
  - [x] Add overwrite option
  - [x] Add verbose option
  - [x] Alert user of unsupported multiple files processing
  - [x] Alert user of missing input and output file extensions and/or formats
  - [x] Alert user if input and/or output formats are out of scope
  - [x] Alert user that input and output formats must not be the same
  - [x] Alert user to use the --overwrite option if file already exists
  - [ ] Test `(--validate)` schema option `(can only be tested during schema generation)`
- [ ] Generate-Converter [#7](https://github.com/usnistgov/oscal-cli-nodejs#7)
  - [ ] Update parameters in the Saxon JS API to perform the converters generation transformation
  - [x] Make module consistent with other modules
  - [x] Allow user to input (XML), and a specified output to produce the JSON (.xsl) or XML (.xsl) converter
  - [x] Save output to user specified location
  - [x] Add fromFormat and toFormat options
  - [x] Add an option with mandatory input values (xml-json, json-xml) to determine which stylesheet is selected
  - [x] Add overwrite option
  - [x] Add verbose option
  - [x] Alert user of unsupported multiple files processing
  - [x] Alert user of missing input and output file extensions and/or formats
  - [x] Alert user if input and/or output formats are out of scope
  - [x] Alert user that input and output formats must not be the same
  - [x] Alert user to use the --overwrite option if file already exists
- [ ] Validate-Schema [#7](https://github.com/usnistgov/oscal-cli-nodejs#7)
  - [x] Make module consistent with other modules
  - [x] Implement AJV and LIBXMLJS2 for schema validation
  - [ ] Change from LIBXMLJS2 to LIBXMLJS - See Issue [oscal-cli-nodejs#10](https://github.com/usnistgov/oscal-cli-nodejs/issues/10)
  - [x] Allow user to input a JSON or XML schema for validation
  - [x] Add json-schema-schema.json validator to validate JSON schema
  - [x] Add verbose option
  - [x] Alert user of unsupported multiple files processing
  - [x] Alert user of missing input file extension and/or format
  - [x] Alert user if input format is out of scope
- [x] Validate-Content [#5 ](https://github.com/usnistgov/oscal-cli-nodejs#5)
  - [x] Implement validation for YAML
  - [x] Alert user if input format is out of scope
  - [x] Test (--validate) support option with the resolve-profile module
- [x] Minify-Json [#5 ](https://github.com/usnistgov/oscal-cli-nodejs#5)
  - [x] Alert user if input and/or output formats are out of scope
- [x] Resolve-Profile [#5 ](https://github.com/usnistgov/oscal-cli-nodejs#5)
  - [x] Update parameters in the Saxon JS API to perform the resolve profile transformation
  - [x] Add support for validating the produced catalog `--validate-result`
- [x] ReadMe.md
  - [x] Update file content - `(IN PROCESS)`
- [ ] Pipeline utility
  - [ ] Generate github secret token for `publishing 'oscalcli tool` 
- [x] Setup-data utility (feedback provided)
  - [x] Remove unnecessary stylesheets
  - [x] Add json-schema-schema.json to build script/folder (Team need to add file to the metaschema repo (production))
  - [x] Replace Linux commands with NodeJS commands to address cross-platform portability issue
- [x] More oscal-cli tool demo feedback for current/future implementation:
  - [x] Remove ./ that precedes the output file (example: \src/./xmltoJSON.json
  - [x] Change the verbose display - provide user with information on current events - (example "Converting JSON to XML ....")

**Comment by avdouglas-nist** [comment]
My status for week ending April 20:

- Completed revising the toFormat and fromFormat options on most of the modules

- Changed Warning messages to Error messages where applicable

- Continue to make minor enhancements throughout the tool

**Comment by aj-stein-nist** [comment]
I have pulled in usnistgov/oscal-cli-nodejs#8. I made some testing and local dev env touchups and will work through current outstanding issues and collab with Dave and Arminta starting tomorrow.

**Comment by aj-stein-nist** [comment]
Met with @david-waltermire-nist and @Arminta-Jenkins-NIST and got her set up on using the current `main` branch code after merging the above PR and will we continue to collaborate on how to complete work needed before release. Per discussion with Dave, I recommend first low-effort item we can tackle with Arminta is https://github.com/usnistgov/oscal-cli-nodejs/issues/12.

Some of the commands below are blocked by me re entity resolution, see above comment about https://github.com/usnistgov/oscal-cli-nodejs/issues/9 and other issues determined while wrapping up testing of the PR.

- generate-schema:  👎 this will not work because it is blocked for a few issues, https://github.com/usnistgov/oscal-cli-nodejs/issues/9 and other things I encountered.
- generate-converters: 👍 I fixed some issues I had and merged them into main
- validate-schema: 🤷 I encountered some weird errors here and need more time to analyze
- convert-format: 👍 this works pretty well
- resolve-profile: 👍 works but the pathing and stuff needs to be reviewed more thoroughly by the group
- validate-content: 🤷 I encountered some weird errors here and need more time to analyze
- minify-json: Dave had confirmed prior this ought to be removed

I need to work with Dave and Arminta to get more issues on the board to accurately track some of these issues.
